packages/itwinui-react/src/core/FileUpload/FileUploadTemplate.test.tsx

```diff
@@ -56,3 +56,34 @@ it('should accept input props', () => {
   expect(input.multiple).toBeFalsy();
   expect(input.accept).toEqual('.txt, .png');
 });
+
+it('should render FileUploadTemplate after a file is uploaded', () => {
+  const mockedOnChange = jest.fn();
+  const file = new File(['mock-file'], 'test.txt', { type: 'text/plain' });
+
+  const { container } = render(
+    <FileUploadTemplate
+      onChange={mockedOnChange}
+      acceptMultiple={false}
+      data={{ file: file }}
+    />,
+  );
+
+  const label = container.querySelector(
+    '.iui-file-uploaded-template-label',
+  ) as HTMLElement;
+  expect(label).toBeTruthy();
+  expect(label.textContent).toEqual('test.txt');
+
+  const secondary = container.querySelector(
+    '.iui-file-uploaded-template-description',
+  ) as HTMLElement;
+  expect(secondary).toBeTruthy();
```

Comment:
If you want to test that the text of the description is being formatted correctly, you could mock `toBytes` and `toDate`.